packages/langium/src/validation/document-validator.ts

```diff
@@ -168,3 +178,15 @@ export function toDiagnosticSeverity(severity: 'error' | 'warning' | 'info' | 'h
             throw new Error('Invalid diagnostic severity: ' + severity);
     }
 }
+
+export namespace DocumentValidator {
+    export const LexingError = 'lexing-error';
+    export const ParsingError = 'parsing-error';
+    export const LinkingError = 'linking-error';
+}
```

Comment:
Since the `code` of diagnostics appears in the frontend, we should have something like a `ValidationCodeProvider` which can be easily overriden by users. They likely want to have more control over which code exactly they want to show. But that's something for another PR.